### ISSUE TITLE
Remove hkdf.c

### ISSUE THREAD

**Issue by gilles-peskine-arm** [Mbed-TLS/mbedtls#9150]
Remove the legacy HKDF implementation (`hkdf.h`, `hkdf.c`, `test_suite_hkdf.*`). PSA has its own independent implementation and no module uses the legacy implementation.

**Comment by MaJerle** [comment]
If you have a customer key exchange protocol, hkdf module is still handy, no?

**Comment by gilles-peskine-arm** [comment]
@MaJerle We aren't removing HKDF. For historical reasons, Mbed TLS has two implementations of HKDF, and we're removing the legacy one.

**Comment by diopoex** [comment]
is that issue still valid? 
see:
`:~/code/mbedtls# find . -name "*hkdf*"
./programs/tf-psa-crypto/tests/CMakeFiles/test_suite_hkdf.dir
./tests/psa-client-server/psasim/src/aut_psa_hkdf.c
./tf-psa-crypto/tests/suites/test_suite_hkdf.data
./tf-psa-crypto/tests/suites/test_suite_hkdf.function
./tf-psa-crypto/tests/test_suite_hkdf
./tf-psa-crypto/tests/test_suite_hkdf.c
./tf-psa-crypto/tests/test_suite_hkdf.datax
./tf-psa-crypto/drivers/builtin/include/mbedtls/hkdf.h
./tf-psa-crypto/drivers/builtin/src/hkdf.o
./tf-psa-crypto/drivers/builtin/src/hkdf.c
./.git/refs/heads/hkdf-remove
./.git/logs/refs/heads/hkdf-remove`

**Comment by gilles-peskine-arm** [comment]
@diopoex Yes, this issue is still relevant (almost entirely in the TF-PSA-Crypto repository now, although there's a tiny bit of cleanup to do in `scripts/data_files/query_config.fmt` in mbedtls).

**Comment by gilles-peskine-arm** [comment]
@mpg I'm changing this issue's priority to MUST because if we don't do it, we need to make `hkdf.h` private as part of https://github.com/Mbed-TLS/mbedtls/issues/10087, and that's not easier than removing it altogether.
